src/main/java/net/mcreator/java/DeclarationFinder.java

```diff
@@ -123,8 +123,8 @@ private static TypeDeclaration getLatestChild(TypeDeclaration parent, int caret)
 		}
 	}
 
-	private static boolean isValidSeparatorContained(String code) {
-		Pattern r = Pattern.compile("[^a-zA-Z_$]");
+	private static boolean isValidSeparatorContained(String code, boolean behind) {
+		Pattern r = behind ? Pattern.compile("[^a-zA-Z0-9_$.]") : Pattern.compile("[^a-zA-Z0-9_$]");
```

Comment:
I suggest you move compiled patterns to public static final variable so they don't get compiled each time this is called.